src/architect/account/account.utils.ts

```diff
@@ -71,6 +72,17 @@ export default class AccountUtils {
       return this.getLocalAccount();
     }
 
+    if (options?.is_component_account_name) {
```

Comment:
I dont think this belongs here. The general idea is valid. However this adds a lot of complexity to an already complex function.

So instead what we should do is create a function that validates if the user is part of the account and then in the command call that to understand if its valid. So in the called class you would have.

```
const component_account_name = ....
const is_valid_account = AccountUtils.isValidAccount(component_account_name);
if (!is_valid_account) {
// Display message here about account not being right.
}
const account = is_valid_account ? component_account_name : flags.account;
// Pass the above var into the account function
```